proxy/src/scram/messages.rs

```diff
@@ -14,7 +14,7 @@ pub const SCRAM_RAW_NONCE_LEN: usize = 18;
 fn validate_sasl_extensions<'a>(parts: impl Iterator<Item = &'a str>) -> Option<()> {
     for mut chars in parts.map(|s| s.chars()) {
         let attr = chars.next()?;
-        if !('a'..='z').contains(&attr) && !('A'..='Z').contains(&attr) {
+        if !attr.is_ascii_lowercase() && !attr.is_ascii_uppercase() {
```

Comment:
Maybe
```suggestion
        if !attr.is_ascii_alphabetic() {
```

Reply:
Hehe, surprise additional ascii method. This would had been probably picked up by $next_version lint :)